lib/ansible/cli/doc.py

```diff
@@ -172,112 +186,187 @@ def display_plugin_list(self, results):
+                        PB_LOADED[pobj] = getattr(loaded_class, pobj, None)
 
-        # add to plugin paths from command line
-        basedir = context.CLIARGS['basedir']
-        if basedir:
-            AnsibleCollectionConfig.playbook_paths = basedir
-            loader.add_directory(basedir, with_subdir=True)
+                    if keyword in PB_LOADED[pobj]._valid_attrs:
+                        kdata['applies_to'].append(pobj)
 
-        if context.CLIARGS['module_path']:
-            for path in context.CLIARGS['module_path']:
-                if path:
-                    loader.add_directory(path)
+                        # we should only need these once
+                        if 'type' not in kdata:
 
-        # save only top level paths for errors
-        search_paths = DocCLI.print_paths(loader)
-        loader._paths = None  # reset so we can use subdirs below
+                            fa = getattr(PB_LOADED[pobj], '_%s' % keyword)
+                            if getattr(fa, 'private'):
+                                kdata = {}
+                                raise KeyError
 
-        # list plugins names or filepath for type, both options share most code
-        if context.CLIARGS['list_files'] or context.CLIARGS['list_dir']:
+                            kdata['type'] = getattr(fa, 'isa', 'string')
 
-            coll_filter = None
-            if len(context.CLIARGS['args']) == 1:
-                coll_filter = context.CLIARGS['args'][0]
+                            if keyword.endswith('when'):
+                                kdata['template'] = 'implicit'
+                            elif getattr(fa, 'static'):
+                                kdata['template'] = 'static'
+                            else:
+                                kdata['template'] = 'explicit'
 
-            if coll_filter in ('', None):
-                paths = loader._get_paths_with_context()
-                for path_context in paths:
-                    self.plugin_list.update(
-                        DocCLI.find_plugins(path_context.path, path_context.internal, plugin_type))
+                            # those that require no processing
+                            for visible in ('alias', 'priority') :
+                                kdata[visible] = getattr(fa, visible)
 
-            add_collection_plugins(self.plugin_list, plugin_type, coll_filter=coll_filter)
+            except KeyError as e:
+                display.warning("Skipping Invalid keyword '%s' specified: %s" % (keyword, to_native(e)))
 
-            # get appropriate content depending on option
-            if context.CLIARGS['list_dir']:
-                results = self._get_plugin_list_descriptions(loader)
-            elif context.CLIARGS['list_files']:
-                results = self._get_plugin_list_filenames(loader)
+            # remove None keys
+            for k in list(kdata.keys()):
```

Comment:
If an unknown keyword is specified, `kdata` has not yet been assigned at this point.

Reply:
descs[keyword] would have issued KeyError already so this would not execute